Group Quota GA API version

tejasm-microsoft opened the pull request:
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

![spec_pr_review_workflow_diagram](https://github.com/Azure/azure-rest-api-specs/assets/4429827/5bb5e7ce-8aff-4dbb-a3f8-0d9b68fef5b1)

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [x] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

tejasm-microsoft committed:
adds the api spec for the GA group quota product

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[Microsoft.Quota](https://apiview.dev/Assemblies/Review/68231935e236449991c31c50e5145e97?revisionId=0b7c5987d6b14c5da159754fbb922249)

tejasm-microsoft committed:
updates readMe for unreferenced error

tejasm-microsoft committed:
adds ready only attribute to the provisioningState for groupQuotaSubscription and GroupQuotaSubscriptionRequest

tejasm-microsoft committed:
modifies examples to have the most recent api-version

tejasm-microsoft committed:
resolves model validation issues

tejasm-microsoft committed:
removes allOf use in provisioningState

mentat9 commented:
@tejasm-microsoft - As far as I can tell, this is a new API version, not updating an existing API version as you indicated in the submission template. Can you clarify the purpose of this PR? If it's really updating an existing API version, the changes should be file modifications, not additions.

tejasm-microsoft commented:
> @tejasm-microsoft - As far as I can tell, this is a new API version, not updating an existing API version as you indicated in the submission template. Can you clarify the purpose of this PR? If it's really updating an existing API version, the changes should be file modifications, not additions.

Sorry, let me reedit my comment. **This is a new version of the API.**The purpose of this PR is to release a stable public version of this API. After our public preview testing, we wanted to remove unnecessary fields in the request body and remove api paths that wont be supported in the public release.

tejasm-microsoft committed:
Merge branch 'main' into group-quota-public-api

rkmanda commented:
> > @tejasm-microsoft - As far as I can tell, this is a new API version, not updating an existing API version as you indicated in the submission template. Can you clarify the purpose of this PR? If it's really updating an existing API version, the changes should be file modifications, not additions.
> 
> Sorry, let me reedit my comment. **This is a new version of the API.** The purpose of this PR is to release a stable public version of this API. After our public preview testing, we wanted to remove unnecessary fields in the request body and remove api paths that wont be supported in the public release.

![image](https://github.com/user-attachments/assets/2fa4b9cb-0496-4708-b1a9-502e4b465211)

Without this we cannot tell what the real difference between your preview and stable versions are. Please create 2 commits that tell us the clear difference

tejasm-microsoft committed:
Commit 1: exact copy of preview version 2024-10-15-preview

tejasm-microsoft committed:
Merge branch 'group-quota-public-api' of https://github.com/Azure/azure-rest-api-specs into group-quota-public-api

tejasm-microsoft committed:
Commit 2 : changes introduced in stable version 2024-12-12

tejasm-microsoft commented:
> > > @tejasm-microsoft - As far as I can tell, this is a new API version, not updating an existing API version as you indicated in the submission template. Can you clarify the purpose of this PR? If it's really updating an existing API version, the changes should be file modifications, not additions.
> > 
> > 
> > Sorry, let me reedit my comment. **This is a new version of the API.** The purpose of this PR is to release a stable public version of this API. After our public preview testing, we wanted to remove unnecessary fields in the request body and remove api paths that wont be supported in the public release.
> 
> ![image](https://private-user-images.githubusercontent.com/16844618/396353905-2fa4b9cb-0496-4708-b1a9-502e4b465211.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ0NTMxMzAsIm5iZiI6MTczNDQ1MjgzMCwicGF0aCI6Ii8xNjg0NDYxOC8zOTYzNTM5MDUtMmZhNGI5Y2ItMDQ5Ni00NzA4LWIxYTktNTAyZTRiNDY1MjExLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjE3VDE2MjcxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWViNDE2NDFjNTRkZTNlMzU3MjkxMWFjMDBjZGY5YzUwYjVhMDk2MzcxYWI5YTUxYzI4NGM2M2E3MTc2NjU0ODMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ylAZsM84S_bzZi5SvR3J7TbVjMqVp7-WrcF1sW9A_QE)
> 
> Without this we cannot tell what the real difference between your preview and stable versions are. Please create 2 commits that tell us the clear difference

Just pushed up the two commits. Just to add some more context about the changes in the stable version. We removed 2 fields in the request body for the Group Quota Request as we noticed that they were unused. We also removed API endpoints that are still being tested in preview but are not ready to be released in the public stable version
